### PR TITLE
TS 4 (again)

### PR DESCRIPTION
**Pull request by slowbackspace**
### Welcome to Typescript 4.0 🎉 🎉 🎉 

tldr; 
- bumped bunch of deps related to TS/ESlint/babel (23 DEPS!!!)
- few `@ts-ignore` added (mostly because we are deleting some prop from an object where the prop is required in TS)
- added fixes related to new eslint rules, one rule responsible for checking camelCase was renamed and bunch of fixes for issues that simply were not caught before such as unused component props. ESlint now discourages use of `Function` type, warns before using `{}` as a type, etc..
- added few TODOs for things that probably could get improved in the future (b74f810b4904b3f35c67d6ea1cb0221175918679)

### CODE REVIEW

**Comment by slowbackspace** [comment]
CI is dead again, in the meantime could you @keraf, @szymonlesisz take a look at these `@ts-ignore`s I added (82d1cb1d4b7a0c3e51fdd73a8e9ef4316b6c3469), if there is something that should really get fixed or we can live with them (few are in tests, but bunch of them in trezorConnectActions, blockchain-link and sendFormBitcoinActions).

**Comment by slowbackspace** [comment]
Finally fixed (properly or by disabling a rule) all the things. If any brave soul that will go through changes and find something that was fixed by disabling rule and he knows how to fix it properly then please feel free to contribute :) 

(I will do a rebase to resolve a conflict afterwards)

File: packages/suite/src/views/wallet/transactions/components/TransactionList/components/TransactionItem/components/TargetAddressLabel/index.tsx [hunk 1]
```diff
@@ -19,6 +19,7 @@ const TargetAddressLabel = ({ target, type }: Props) => {
         <Translation id="TR_SENT_TO_SELF" />
     ) : (
         target.addresses?.map((a, i) =>
+            // eslint-disable-next-line react/no-array-index-key
```
Comment (tomexx) [inline packages/suite/src/views/wallet/transactions/components/TransactionList/components/TransactionItem/components/TargetAddressLabel/index.tsx:23]: Doesn't address have some id that could be used here?

Comment (slowbackspace) [reply packages/suite/src/views/wallet/transactions/components/TransactionList/components/TransactionItem/components/TargetAddressLabel/index.tsx:23]: not really, there is only address itself. I assume that tx can have multiple outputs going to the same address. Am I correct @szymonlesisz?

Anyway using index is fine in this case as the array is always the same (we are not changing its items or reordering it). some reading material about it (scroll to "Exception from the rule") https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318

File: packages/components/src/components/Modal/index.tsx [hunk 2]
```diff
@@ -67,44 +66,42 @@ type ModalWindowProps = Omit<
...
-  }
+        `${props.modalPaddingTop[3]} ${props.modalPaddingSide[3]} ${props.modalPaddingBottom[3]}`};
 
-  @media only screen and (min-width: ${variables.SCREEN_SIZE.SM}) and (max-width: ${
-    variables.SCREEN_SIZE.MD
-}) {
-    padding: ${(props: ModalWindowProps) =>
-        `${props.modalPaddingTop[1]} ${props.modalPaddingSide[1]} ${props.modalPaddingBottom[1]}`};
-  }
+    /* prettier fails to format it properly */
```
Comment (szymonlesisz) [inline packages/components/src/components/Modal/index.tsx]: rm

File: packages/suite/src/services/suite/invityAPI.ts [hunk 3]
```diff
@@ -71,7 +71,17 @@ class InvityAPI {
         this.server = server;
     }
 
-    private options(body = {}, method = 'POST'): any {
+    private options(
+        body:
```
Comment (szymonlesisz) [inline packages/suite/src/services/suite/invityAPI.ts]: share type in both methods

File: packages/suite/src/utils/suite/__fixtures__/device.ts [hunk 4]
```diff
@@ -348,7 +348,7 @@ const d = (obj: any) => ({
     path: obj.path ? obj.path : obj.id,
     features: obj.id
         ? {
-              // eslint-disable-next-line @typescript-eslint/camelcase
+              // eslint-disable-next-line @typescript-eslint/naming-convention
```
Comment (szymonlesisz) [inline packages/suite/src/utils/suite/__fixtures__/device.ts]: rm

File: packages/suite/src/utils/suite/analytics.ts [hunk 5]
```diff
@@ -13,15 +13,15 @@ export const encodeDataToQueryString = (event: AnalyticsEvent, common: Common) =
     const { type } = event;
     const { version, sessionId, instanceId } = common;
     const commonEncoded = qs.stringify({
```
Comment (szymonlesisz) [inline packages/suite/src/utils/suite/analytics.ts:15]: eslint-disable @typescript-eslint/naming-convention

File: packages/suite/src/utils/suite/analytics.ts [hunk 6]
```diff
@@ -13,15 +13,15 @@ export const encodeDataToQueryString = (event: AnalyticsEvent, common: Common) =
...
-        // eslint-disable-next-line @typescript-eslint/camelcase
+        // eslint-disable-next-line @typescript-eslint/naming-convention
         c_commit: process.env.COMMITHASH,
-        // eslint-disable-next-line @typescript-eslint/camelcase
+        // eslint-disable-next-line @typescript-eslint/naming-convention
         c_instance_id: instanceId,
-        // eslint-disable-next-line @typescript-eslint/camelcase
+        // eslint-disable-next-line @typescript-eslint/naming-convention
         c_session_id: sessionId,
     });
```
Comment (szymonlesisz) [inline packages/suite/src/utils/suite/analytics.ts:23]: eslint-enable @typescript-eslint/naming-convention

File: packages/suite/src/views/wallet/transactions/components/TransactionList/components/TransactionItem/components/TargetAddressLabel/index.tsx (hunk 1 above)
Comment (szymonlesisz) [reply packages/suite/src/views/wallet/transactions/components/TransactionList/components/TransactionItem/components/TargetAddressLabel/index.tsx:23]: yes its safe :) comment why

**Review by szymonlesisz — CHANGES_REQUESTED** [review]

File: packages/suite/src/components/backup/AfterBackupCheckboxes.tsx [hunk 7]
```diff
@@ -16,7 +16,6 @@ const CheckboxWrapper = styled.div`
 `;
 
 const mapStateToProps = (state: AppState) => ({
-    device: state.suite.device,
```
Comment (slowbackspace) [inline packages/suite/src/components/backup/AfterBackupCheckboxes.tsx:19]: unused prop?

File: packages/suite/src/components/backup/PreBackupCheckboxes.tsx [hunk 8]
```diff
@@ -16,7 +16,6 @@ const CheckboxWrapper = styled.div`
 `;
 
 const mapStateToProps = (state: AppState) => ({
-    device: state.suite.device,
```
Comment (slowbackspace) [inline packages/suite/src/components/backup/PreBackupCheckboxes.tsx:19]: unused prop?

File: packages/suite/src/components/suite/modals/TransactionDetail/index.tsx [hunk 9]
```diff
@@ -43,18 +40,14 @@ const AdvancedDetailsWrapper = styled.div`
     padding: 24px 24px 14px 24px;
 `;
 
-const mapStateToProps = (state: AppState) => ({
-    fiat: state.wallet.fiat,
```
Comment (slowbackspace) [inline packages/suite/src/components/suite/modals/TransactionDetail/index.tsx:47]: unused prop, ok to remove

File: packages/suite/src/components/wallet/AccountException/AccountNotEnabled.tsx [hunk 10]
```diff
@@ -2,14 +2,13 @@ import React from 'react';
...
 import { useDevice, useActions } from '@suite-hooks';
-import { Network, Discovery } from '@wallet-types';
+import { Network } from '@wallet-types';
 import { Translation, Image } from '@suite-components';
 
 import Wrapper from './components/Wrapper';
 
 interface Props {
     network: Network;
-    discovery: Discovery;
```
Comment (slowbackspace) [inline packages/suite/src/components/wallet/AccountException/AccountNotEnabled.tsx:12]: unused prop?

File: packages/suite/src/views/onboarding/unexpected-states/index.tsx [hunk 11]
```diff
@@ -29,7 +29,6 @@ const IsInBootloader = () => (
 const mapStateToProps = (state: AppState) => ({
     onboarding: state.onboarding,
     suite: state.suite,
-    recovery: state.recovery,
```
Comment (slowbackspace) [inline packages/suite/src/views/onboarding/unexpected-states/index.tsx:32]: unused prop?

File: packages/suite/src/views/suite/bridge/index.tsx [hunk 12]
```diff
@@ -87,7 +87,6 @@ const Col = styled.div<{ justify?: string }>`
 
 const mapStateToProps = (state: AppState) => ({
     transport: state.suite.transport,
-    device: state.suite.device,
```
Comment (slowbackspace) [inline packages/suite/src/views/suite/bridge/index.tsx:90]: unused prop? imo safe to remove

File: packages/suite/src/views/suite/device-connect/index.tsx [hunk 13]
```diff
@@ -35,7 +35,6 @@ const StyledLink = styled(Link)``;
 
 const mapStateToProps = (state: AppState) => ({
     transport: state.suite.transport,
-    debug: state.suite.settings.debug,
```
Comment (slowbackspace) [inline packages/suite/src/views/suite/device-connect/index.tsx:38]: unused prop, safe to remove

File: packages/suite/src/components/backup/PreBackupCheckboxes.tsx (hunk 8 above)
Comment (slowbackspace) [reply packages/suite/src/components/backup/PreBackupCheckboxes.tsx:19]: imo safe to remove

File: packages/suite/src/components/backup/AfterBackupCheckboxes.tsx (hunk 7 above)
Comment (slowbackspace) [reply packages/suite/src/components/backup/AfterBackupCheckboxes.tsx:19]: imo safe to remove

**Review by szymonlesisz — APPROVED** [review]

**Comment by mroz22** [comment]
# :tada:

**Pull request merged**